src/TestHeaders.am

```diff
@@ -5,21 +5,31 @@
 ## Please see the COPYING and CONTRIBUTORS files for details.
 ##
 
-## Test ./*.h files in a directory using test-suite/testheaders.sh
+# A space-separated list of header files to be skipped by
+# testHeaders during "make check"
+EXCLUDE_FROM_HDR_TESTING =
 
-## TODO: include in src/Common.am when all testHeaders users are the same.
 
-TESTS += testHeaders
-
-## .h dependency test script
-## aborts build process on errors; XXX: even with "make -k"
-testHeaders: $(srcdir)/*.h
-	$(SHELL) $(top_srcdir)/test-suite/testheaders.sh "$(CXXCOMPILE)" $^ || exit 1
+# to test additional header files to those tested by default,
+# add an empty target named 'testHeaders' depending on the additional
+# headers to test
 
-## XXX: this is only needed because testheaders.sh creates a dummy file called
-## testHeaders and distclean does not know about it.
-## Remove when testheaders.sh and its users are fixed.
-CLEANFILES += testHeaders
+testHeaders: $(SOURCES) $(noinst_HEADERS) $(EXTRA_DIST) $(top_srcdir)/test-suite/testHeader.cc.in
+	$(MAKE) $(^:.h=.hdrtest) && cp $(TRUE) $@
```

Comment:
This PR does not fix bug [5237](https://bugs.squid-cache.org/show_bug.cgi?id=5237#c1) AFAICT -- we are still copying a busybox binary (link) here -- so I have removed that claim from the PR title. If I am wrong regarding the lack of a fix, then I would still not put bug 5237 fix in _title_ of this PR because this PR is not really about that bug.

This comment is not a change request and can be resolved as long as that bug reference is kept away from this PR title.

Reply:
We could build a "true" shell script there... let me try that

Reply:
Yes, we can fix that bug, but, IMO, we should not fix that bug in _this_ PR!

Reply:
proposed fix to that bug is attached to the bug